extensions/data-plane/data-plane-aws-s3/src/main/java/org/eclipse/edc/connector/dataplane/aws/s3/S3DataSink.java

```diff
@@ -91,16 +98,23 @@ protected StreamResult<Object> transferParts(List<DataSource.Part> parts) {
         return StreamResult.success();
     }
 
-    String getDestinationObjectName(String partName) {
+    private String getDestinationObjectName(String partName) {
         if (!StringUtils.isNullOrEmpty(folderName)) {
             return folderName.endsWith("/") ? folderName + partName : folderName + "/" + partName;
         }
         return partName;
     }
 
+    @NotNull
+    private StreamResult<Object> downloadFailure(Exception e, String keyName) {
+        var message = format("Error downloading the %s object: %s", keyName, e.getMessage());
+        monitor.severe(message, e);
+        return StreamResult.error(message);
+    }
+
     @NotNull
     private StreamResult<Object> uploadFailure(Exception e, String keyName) {
-        var message = format("Error writing the %s object on the %s bucket: %s", keyName, bucketName, e.getMessage());
+        var message = format("Error uploading the %s object: %s", keyName, e.getMessage());
         monitor.severe(message, e);
         return StreamResult.error(message);
     }
```

Comment:
these two metohds are the same except for the `uploading/downloading` word, please extract a single method